### PR TITLE
Migrate kdm, chart, system chart branch values to the new build system

### PR DESCRIPTION
**Pull request by maxsokolovsky**
One other change we need to check is the removal of `ENV CATTLE_KDM_BRANCH=dev-v2.8` from Dockerfile.dapper.

This line is continually updated - the KDM branch.
https://github.com/rancher/rancher/blob/79bf51eb1ce0570915b18669a14044c2378dbfa3/Dockerfile.dapper#L16C31-L16C31
But looks like we later update it elsewhere - in `package/Dockerfile` and some of the scripts. It appears we may not need to set the value in Dockerfile.dapper.

We're trying to integrate the KDM branch with the new build system, so that it's not specified inline, but taken from outside as an ARG. Where do we explicitly run docker build with this dapper file?

This also updates the `check-chart-kdm-source-values` script for 2.9.

### CODE REVIEW

File: build.yaml [hunk 1]
```diff
@@ -1,4 +1,9 @@
+# To understand how to use the build/config system, read the docs in docs/build.md.
+
 webhookVersion: 104.0.0+up0.5.0-rc6
 cspAdapterMinVersion: 103.0.0+up3.0.0
 defaultShellVersion: rancher/shell:v0.1.22
+systemChartDefaultBranch: dev-v2.9
+chartDefaultBranch: dev-v2.9
+kdmBranch: dev-v2.9
```
Comment (jiaqiluo) [inline build.yaml:8]: Question: is it true that we will need to update those values at two places: this file and the file `pkg/buildconfig/constants.go` ?

**Comment by jiaqiluo** [comment]
As the changes in the build system will also affect the releasing of rancher-prime, we would like to have the release team review this PR as well.  cc @briandowns

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: build.yaml (hunk 1 above)
Comment (maxsokolovsky) [reply build.yaml:8]: The constants file is for use of these values in any Go code (if and when needed). But you need to change only the `build.yaml` file. Then you run `go generate`, and the tool updates the constants file for you. But changes to both files need to be committed.

**Review by jiaqiluo — APPROVED** [review]
LGTM

**Review by snasovich — COMMENTED** [review]
@maxsokolovsky , leaving the actual PR review up to others but please ensure that these changes "play nicely" with whatever process generates `Chart/KDM sources` on autogenerated RC release notes like https://github.com/rancher/rancher/releases/tag/v2.8.3-rc7 - not sure if we need any corresponding updates there.

**Comment by maxsokolovsky** [comment]
@snasovich, thanks for the note, I didn't know about that, will check it out!

**Comment by maxsokolovsky** [comment]
@briandowns, I updated the `check-chart-kdm-source-values` script for 2.9 and the new build system.
One thing to note is that the `create-components-file.sh` script creates the listing of chart/KDM sources with the variable names as they are written in the build.yaml file itself. It doesn't convert them to ENV_VAR case. Please let me know if that's OK. The output currently looks like this:
```
...
# Chart/KDM sources
* systemChartDefaultBranch: dev-v2.9 (`build.yaml`)
* chartDefaultBranch: dev-v2.9 (`build.yaml`)
* kdmBranch: dev-v2.9 (`build.yaml`)
```

**Pull request closed without merging**